### PR TITLE
Add hanging_place event

### PR DESCRIPTION
**Pull request by Lildirt**

### CODE REVIEW

File: src/main/java/com/laytonsmith/core/events/drivers/EntityEvents.java [hunk 1]
```diff
@@ -1856,6 +1857,81 @@ public boolean modifyEvent(String key, Mixed value, BindableEvent event) {
...
+			return MSVersion.V3_3_4;
+		}
+
+		@Override
+		public boolean matches(Map<String, Mixed> prefilter, BindableEvent event) throws PrefilterNonMatchException {
+			if(event instanceof MCHangingPlaceEvent) {
+				MCHangingPlaceEvent hangingPlaceEvent = (MCHangingPlaceEvent) event;
+				MCHanging hanging = hangingPlaceEvent.getEntity();
+				Prefilters.match(prefilter, "type", hanging.getType().name(), PrefilterType.MACRO);
+				Prefilters.match(prefilter, "id", hanging.getUniqueId().toString(), PrefilterType.MACRO);
```
Comment (PseudoKnight) [inline src/main/java/com/laytonsmith/core/events/drivers/EntityEvents.java:1891]: I'm pretty sure the code will never know the id of the placed entity. So this prefilter is unnecessary.

File: src/main/java/com/laytonsmith/core/events/drivers/EntityEvents.java [hunk 2]
```diff
@@ -1856,6 +1857,81 @@ public boolean modifyEvent(String key, Mixed value, BindableEvent event) {
...
+		@Override
+		public Map<String, Mixed> evaluate(BindableEvent event) throws EventException {
+			if(event instanceof MCHangingPlaceEvent) {
+				MCHangingPlaceEvent hangingPlaceEvent = (MCHangingPlaceEvent) event;
+				Map<String, Mixed> ret = evaluate_helper(event);
+				MCHanging hanging = hangingPlaceEvent.getEntity();
+
+				ret.put("type", new CString(hanging.getType().name(), Target.UNKNOWN));
+				ret.put("id", new CString(hanging.getUniqueId().toString(), Target.UNKNOWN));
+				ret.put("player", new CString(((MCHangingPlaceEvent) event).getPlayer().getName(), Target.UNKNOWN));
```
Comment (PseudoKnight) [inline src/main/java/com/laytonsmith/core/events/drivers/EntityEvents.java:1915]: Small thing, but you already casted the hangingPlaceEvent above.

File: src/main/java/com/laytonsmith/core/events/drivers/EntityEvents.java [hunk 3]
```diff
@@ -1856,6 +1857,81 @@ public boolean modifyEvent(String key, Mixed value, BindableEvent event) {
...
+		}
+
+		@Override
+		public String docs() {
+			return "{type: <macro> The entity type of the entity | id: <macro> The entity ID of the entity"
+					+ " | player: <macro> The player triggering the event | world: <macro> The world the entity is in}"
+					+ " This event is called when a player attempts to place an item frame, painting, or leash."
+					+ " {id: The entity ID of the entity. | type: The type of the hanging entity; can be ITEM_FRAME,"
+					+ " PAINTING, or LEASE_HITCH | location: Where the entity was placed. |"
+					+ " player: The name of the place which placed this entity. }"
```
Comment (PseudoKnight) [inline src/main/java/com/laytonsmith/core/events/drivers/EntityEvents.java:1875]: typo: "name of the player"

**Comment by Lildirt** [comment]
I've made the amendments requested.

**Comment by PseudoKnight** [comment]
Looks good. Squash commits?

**Comment by Lildirt** [comment]
Squashed.

**Pull request merged**